src/blocks/food-and-drinks/food-item/edit.js

```diff
@@ -89,13 +89,17 @@ const FoodItem = ( props ) => {
 		if ( showImage !== prevShowPrice ) {
 			setAttributes( { url: showImage ? url : '' } );
 		}
+
+		if ( !! url && url !== attributes.url ) {
+			setAttributes( { url } );
+		}
```

Comment:
For posterity: This change was needed because there were situations where the url attribute had been undefined.